### ISSUE TITLE
Fix S3241 FN: when the unused return value is a method parameter

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#3247]
### Description

See this [community thread.](https://community.sonarsource.com/t/rule-s3241-fn-methods-should-not-return-values-that-are-never-used-code-smell-warning-is-not-caught-by-sonar-scanner-and-it-is-not-displayed-in-the-report-on-sonarqube-8-2-version-community-edition/21885)

A private method can return one of its parameters. As long as this returned value is not used, S3241 should raise an issue.

### Repro steps

```
    public class UnusedValueIsParameter
    {
        private int NeverUsed(int neverUsedVal) // FN - the returned value is not used
        {
            return neverUsedVal;
        }
    }

```

### Related information

* SonarC# Version 8.5

**Comment by pavel-mikula-sonarsource** [comment]
We'll need to be careful about methods implementing interface here.
```C#
private int AdjustSize(Size size){
  return size;
}
```

**Comment by andrei-epure-sonarsource** [comment]
Good catch @csaba-sagi-sonarsource , and indeed from [the user's reply](https://community.sonarsource.com/t/rule-s3241-fn-methods-should-not-return-values-that-are-never-used-code-smell-warning-is-not-caught-by-sonar-scanner-and-it-is-not-displayed-in-the-report-on-sonarqube-8-2-version-community-edition/21885/5?u=andrei_epure) the rule is working well.

I opened #5511 to fix the tests.
